docs/src/pages/components/data-grid/columns/ColumnSelectorGrid.tsx

```diff
@@ -11,7 +11,12 @@ export default function ColumnSelectorGrid() {
 
   return (
     <div style={{ height: 400, width: '100%' }}>
-      <DataGrid {...data} showToolbar />
+      <DataGrid
+        {...data}
+        components={{
+          Header: GridHeader,
```

Comment:
I was expecting the usage of:

```jsx
<GridToolbar>
  <GridToolbarColumnSelector />
</GridToolbar>
```

as a header for this demo since it's specific to the column selector.  The same logic would apply for the specific demos of filtering and density. What do you think? 

I guess the usage of `GridHeader` can make sense for the developers that want it all, but I'm not sure that it's common 🤔

Reply:
This is to make the docs work for now. We should discuss how to handle those examples. We can have the ones we currently have as they are but add another one that is specific for composing the toolbar. We can put it in the `rendering` section.

Reply:
If we keep the pull request as follows. What's the value of the change?

Reply:
Sure, I can make custom toolbars per demo (for example the one for the density will only have the density selector in the toolbar). I make the current version to have a baseline to start discussing how the docs will look like.

Reply:
Updated